### PR TITLE
Only reset settings if dbconfig specified

### PR DESCRIPTION
**Pull request by lmcarril**
Avoid to call reset_config and apply_config if the user does not specify any setting for the database.
That allows the benchmark to be run by users without superuser privileges. e.g. in AWS RDS the user cannot be a real superuser).

Also clarify in the documentation that being superuser is required, and that the reversal of the settings means going back to the ones defined in the configuration file.

### CODE REVIEW

File: s64da_benchmark_toolkit/streams.py [hunk 1]
```diff
@@ -81,12 +81,14 @@ def apply_sql_modifications(sql, modifications):
...
 
     def run(self):
+        dbconfig = self.config.get('dbconfig')
         try:
             mp_manager = Manager()
             reporting_queue = mp_manager.Queue()
 
-            self.db.reset_config()
-            self.db.apply_config(self.config.get('dbconfig', {}))
+            if dbconfig != None:
```
Comment (sdressler) [inline s64da_benchmark_toolkit/streams.py:89]: You can shorten it to `if dbconfig:`

File: s64da_benchmark_toolkit/streams.py [hunk 2]
```diff
@@ -96,7 +98,8 @@ def run(self):
             pass
 
         finally:
-            self.db.reset_config()
+            if dbconfig != None:
```
Comment (sdressler) [inline s64da_benchmark_toolkit/streams.py:101]: Same here: `if dbconfig:`

**Comment by jukoson** [comment]
Since we don't have the functionality yet in teamcity to run a benchmark on this branch - have you verified it runs ok for tpch 10 ?

File: s64da_benchmark_toolkit/streams.py (hunk 2 above)
Comment (lmcarril) [reply s64da_benchmark_toolkit/streams.py]: done

File: s64da_benchmark_toolkit/streams.py (hunk 1 above)
Comment (lmcarril) [reply s64da_benchmark_toolkit/streams.py]: done

**Comment by lmcarril** [comment]
> Since we don't have the functionality yet in teamcity to run a benchmark on this branch - have you verified it runs ok for tpch 10 ?

will do

**Comment by lmcarril** [comment]
Manually tested the following and checked the postgres log to observe if the configuration where applied and reset as expected:

- Without specifying config file, the benchmark runs without calling reset_config and apply_config
- If a config file is specified but  it contains no dbconfig,  the benchmark runs without calling reset_config and apply_config
- If the config file is specified and there is a dbconfig section, the benchmars runs but the config is applied and reset as expected.

So for what I see, the patch works as expected.

**Pull request merged**